### PR TITLE
Add QuadThresholdParameters to AprilTag config

### PR DESCRIPTION
**Pull request by crschardt**
This works around a change made to the default QuadThresholdParameters in the WPILib AprilTagDetector for 2025
https://github.com/wpilibsuite/allwpilib/pull/6847

### CODE REVIEW

File: photon-core/src/main/java/org/photonvision/vision/pipe/impl/AprilTagDetectionPipeParams.java [hunk 1]
```diff
@@ -45,8 +50,27 @@ public boolean equals(Object obj) {
         if (getClass() != obj.getClass()) return false;
         AprilTagDetectionPipeParams other = (AprilTagDetectionPipeParams) obj;
         if (family != other.family) return false;
+
+        boolean sameQuadParams = false;
+        boolean sameDetectorParams = false;
         if (detectorParams == null) {
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/vision/pipe/impl/AprilTagDetectionPipeParams.java:56]: When will these inputs be null? i'm having a hard time following the chain of null/equality checks is all.

File: photon-core/src/main/java/org/photonvision/vision/pipeline/AprilTagPipeline.java [hunk 2]
```diff
@@ -87,7 +87,12 @@ protected void setPipeParamsImpl() {
         config.refineEdges = settings.refineEdges;
         config.quadSigma = (float) settings.blur;
         config.quadDecimate = settings.decimate;
-        aprilTagDetectionPipe.setParams(new AprilTagDetectionPipeParams(settings.tagFamily, config));
+
+        var quadParams = new AprilTagDetector.QuadThresholdParameters();
+        quadParams.minClusterPixels = 5;
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/vision/pipeline/AprilTagPipeline.java:92]: do we not wanna explicitly set the other parameters too?

File: photon-core/src/main/java/org/photonvision/vision/pipe/impl/AprilTagDetectionPipeParams.java (hunk 1 above)
Comment (crschardt) [reply photon-core/src/main/java/org/photonvision/vision/pipe/impl/AprilTagDetectionPipeParams.java:56]: I don't know. I copied the existing logic figuring that it was there for a reason.

Comment (mcm001) [reply photon-core/src/main/java/org/photonvision/vision/pipe/impl/AprilTagDetectionPipeParams.java:56]: For equals/hash code I usually accept my IDE auto code gen as truth. Don't put any stock in the old stuff.

File: photon-core/src/main/java/org/photonvision/vision/pipeline/AprilTagPipeline.java (hunk 2 above)
Comment (crschardt) [reply photon-core/src/main/java/org/photonvision/vision/pipeline/AprilTagPipeline.java:92]: I'm good with setting them all. I did this one first to make sure it fixed the problem.

Comment (mcm001) [reply photon-core/src/main/java/org/photonvision/vision/pipeline/AprilTagPipeline.java:92]: Sweet, let's do that and just drop a comment explaining why we chose the numbers (they seem to work OK)

File: photon-core/src/main/java/org/photonvision/vision/pipe/impl/AprilTagDetectionPipeParams.java (hunk 1 above)
Comment (crschardt) [reply photon-core/src/main/java/org/photonvision/vision/pipe/impl/AprilTagDetectionPipeParams.java:56]: I didn't know that was a thing. I've updated them with the autogenerated ones.

**Review by mcm001 — APPROVED** [review]
Seeeend it

**Pull request merged**